Add OffScreenCanvas context lost/restored events (#33159)
mdn/content: files/en-us/web/api/offscreencanvas/index.md

- {{domxref("OffscreenCanvas.transferToImageBitmap()")}}
  - : Creates an {{domxref("ImageBitmap")}} object from the most recently rendered image of the `OffscreenCanvas`. See the {{domxref("OffscreenCanvas.transferToImageBitmap()", "API description")}} for important notes on managing this {{domxref("ImageBitmap")}}.

## Events

_Inherits events from its parent, {{domxref("EventTarget")}}._

Listen to these events using {{DOMxRef("EventTarget.addEventListener", "addEventListener()")}} or by assigning an event listener to the `oneventname` property of this interface.

- [`contextlost`](/en-US/docs/Web/API/OffscreenCanvas/contextlost_event) {{Experimental_Inline}}

>   - : Fired if the user agent detects that the backing storage associated with an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context has been lost.

Author: ```suggestion
  - : Fired if the browser detects that the underlying system that provides the resources used by the [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context has been lost.
```

>   - : Fired if the browser detects that the underlying system that provides the resources used by an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context, such as a GPU, has been lost.

Reviewer: ```suggestion
  - : Fired if the browser detects that an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost, for example, when a GPU process has crashed.
```

Author: ```suggestion
- : Fired if the browser detects that an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.
```

- [`contextrestored`](/en-US/docs/Web/API/OffscreenCanvas/contextrestored_event) {{Experimental_Inline}}

>   - : Fired if the user agent successfully restores a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context.

Author: ```suggestion
  - : Fired if the user agent successfully restores the underlying system that provides the resources used by an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context.
```

>   - : Fired if the browser successfully restores the underlying system that provides the resources used by an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context.

Reviewer: ```suggestion
  - : Fired if the browser successfully restores an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context, for example, after a GPU process has crashed.
```

Author: ```suggestion
  - : Fired if the browser successfully restores an [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context.
```
